pkg/util/github/secrets_test.go

```diff
@@ -0,0 +1,127 @@
+package github_test
+
+import (
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/devstream-io/devstream/pkg/util/github"
+)
+
+var _ = Describe("Secrets", func() {
+	var owner, repo, org = "o", "r", "or"
+	var registerUrl = fmt.Sprintf("/repos/%v/%v/actions/secrets/public-key", org, repo)
+	sk, sv := "sk", "sv"
+
+	Context(("do AddRepoSecret"), func() {
+		It("step1: do GetRepoPublicKey with wrong url", func() {
+			rightClient, err := github.NewClientWithOption(&github.Option{
+				Owner: owner,
+				Repo:  repo,
+				Org:   org,
+			}, serverURL)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(rightClient).NotTo(Equal(nil))
+			err = rightClient.AddRepoSecret(sk, sv)
+			Expect(err).NotTo(Succeed())
+		})
+
+		It("step2: do AddRepoSecret with correct url", func() {
+			registerUrl = fmt.Sprintf("/repos/%v/%v/actions/secrets/public-key", org, repo)
+			mux.HandleFunc(registerUrl, func(w http.ResponseWriter, r *http.Request) {
+				fmt.Fprint(w, `{"key_id":"1234","key":"2Sg8iYjAxxmI2LvUXpJjkYrMxURPc8r+dB7TJyvv1234"}`)
+			})
+			rightClient, err := github.NewClientWithOption(&github.Option{
+				Owner: owner,
+				Repo:  repo,
+				Org:   org,
+			}, serverURL)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(rightClient).NotTo(Equal(nil))
+			err = rightClient.AddRepoSecret(sk, sv)
+			Expect(err).NotTo(Succeed())
+		})
+
+		It("step3: do CreateOrUpdateRepoSecret with wrong url", func() {
+			rightClient, err := github.NewClientWithOption(&github.Option{
+				Owner: owner,
+				Repo:  repo,
+				Org:   org,
+			}, serverURL)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(rightClient).NotTo(Equal(nil))
+
+			err = rightClient.AddRepoSecret(sk, sv)
+			Expect(err).NotTo(Succeed())
+		})
+
+		It("step3: do CreateOrUpdateRepoSecret with correct url", func() {
```

Comment:
because step3 corresponds to the source file is an `if-else` statement, the 'step3...wrong url' is used to cover the situation of `err != nil`, and the  'step3...correct url' is used to cover the right condition.